src/Compilers/Server/VBCSCompiler/DesktopBuildServerController.cs

```diff
@@ -25,6 +27,12 @@ internal DesktopBuildServerController(NameValueCollection appSettings)
 
         protected override IClientConnectionHost CreateClientConnectionHost(string pipeName)
         {
+            var compilerServerHost = CreateCompilerServerHost();
+            return CreateClientConnectionHostForServerHost(compilerServerHost, pipeName);
+        }
+
+        internal static ICompilerServerHost CreateCompilerServerHost()
+        {
             // VBCSCompiler is installed in the same directory as csc.exe and vbc.exe which is also the 
             // location of the response files.
             var clientDirectory = AppDomain.CurrentDomain.BaseDirectory;
```

Comment:
How is this working on CoreCLR since it doesn't have AppDomains?

Reply:
CoreCLR does have AppDomains - I'm not sure if they're actually implemented in full, but they at least implemented all the APIs that don't really have to do with AppDomains proper (such as this API, getting a directory, don't need special support for that): https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/AppDomain.cs#L332

Reply:
Ah, I see, it has exactly one AppDomain -- which is good enough for our purposes.